src/onehot.jl

```diff
@@ -183,9 +183,27 @@ julia> reshape(1:15, 3, 5) * oh  # this matrix multiplication is done efficientl
  3  6  15  3  9  3  12  3  6  15  3
 ```
 """
-onehotbatch(ls, labels, default...) = _onehotbatch(ls, length(labels) < 32 ? Tuple(labels) : labels, default...)
-# NB function barier:
-_onehotbatch(ls, labels, default...) = batch([onehot(l, labels, default...) for l in ls])
+onehotbatch(data, labels, default...) = _onehotbatch(data, length(labels) < 32 ? Tuple(labels) : labels, default...)
+
+# NB function barrier:
+function _onehotbatch(data, labels)
+  n_labels = length(labels)
+  indices = _findval.(data, Ref(labels))
```

Comment:
I think using `map` is better than broadcasting here. 
```suggestion
  indices = map(x -> _findval(x, labels), data)
```
If `data` is a non-array iterable (like a generator) then this will save collecting it first.

Reply:
Thanks, good shout.